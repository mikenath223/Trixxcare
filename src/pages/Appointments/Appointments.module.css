.container {
  width: 99vw;
  height: 99vh;
  display: flex;
  justify-content: center;
}

.containerLoad {
  background: #E5EFF1;
  align-items: center;
}

.user {
  display: flex;
  margin-top: 10px;
  align-items: center;
  font-size: 30px;
}

.menuIcon, .closeMenu {
  display: none;
}

.menuWrap {
  width: min-content;
}

.midBar p:first-child {
  cursor: pointer;
}

.mainBar {
  width: 80%;
  display: flex;
  padding-top: 20px;
  align-items: center;
  flex-direction: column;
}

.appBar {
  width: 100%;
  height: 500px;
  padding-top: 20px;
  overflow-y: auto;
}

.appBar h3 {
  font-size: 18px;
}

.mainBar>h3 {
  font-size: 30px;
  color: #1a9afcfb;
  margin-bottom: 20px;
}

.table {
  width: 95%;
  margin: auto;
}

.appBar::-webkit-scrollbar {
  width: 10px;
}

.appBar::-webkit-scrollbar-track {
  box-shadow: inset 0 0 6px rgba(0, 0, 0, 1);
  -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 1);
  -webkit-border-radius: 10px;
  border-radius: 10px;
}

.appBar::-webkit-scrollbar-thumb {
  -webkit-border-radius: 10px;
  box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.5);
  -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.5);
  border-radius: 10px;
  background: #1a9afcfb;
}

.tableRow {
  box-shadow: 1px 2px 0 silver;
  display: flex;
  align-items: center;
  justify-content: space-around;
  padding: 10px;
}

.tableData {
  display: flex;
  align-items: center;
  width: 100%;
  text-align: center;
}

.delButton {
  border: none;
  background-color: #1a9afcfb;
  color: #fff;
  padding: 5px;
  font-size: 18px;
  cursor: pointer;
}

.noApp a {
  border-radius: 10px;
  border: 2px solid gray;
  background-color: #99C210;
  color: #fff;
  padding: 5px;
  margin: 7px;
  display: inline-block;
}

@media (max-width: 770px) {
  .sideBar {
    display: none;
  }
  .container, .containerLoad {
    min-height: 100vh;
  }
  .appBar {
    margin-top: 0;
    padding-top: 10px;
    height: auto;
    overflow: initial;
  }
  .mainBar {
    margin-top: 30px;
    width: 95%;
    margin-bottom: 20px;
  }
  .delButton {
    font-size: 16px;
  }
  .menuIcon, .closeMenu {
    display: initial;
    cursor: pointer;
  }
  .menuIcon {
    position: absolute;
    top: 10px;
    left: 10px;
  }
  .closeWrap {
    position: relative;
    left: -40%;
    top: 10px;
  }
  .sideBar {
    position: absolute;
    top: 0;
    left: 0;
    z-index: 5;
    width: 40%;
    background-color: #fff;
  }
  .rightBar {
    width: 65%;
  }
}

@media (max-width: 430px) {
  .delButton {
    font-size: 13px;
  }
  .tableRow {
    flex-direction: column;
    margin-bottom: 10px;
  }
  .tableData {
    padding: 5px 0;
    justify-content: center;
  }
  .sideBar {
    width: 70%;
  }
}