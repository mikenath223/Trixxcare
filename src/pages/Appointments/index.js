import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { Link, useHistory } from 'react-router-dom';
import SweetAlert from 'react-bootstrap-sweetalert';
import PropTypes from 'prop-types';
import { handleCloseMenu, handleOpenMenu, resizer } from '../../utils/domlist';
import { getAppointments, deleteAppoints } from 'utils/request';
import {
  SETAPPOINT, SETLOGOUT, DELAPPOINT,
} from 'store/actions';
import logo from 'assets/images/logo.png';
import Error from 'pages/Error';
import styles from './Appointments.module.css';
import style from 'pages/Dashboard/Dashboard.module.css';


const mapStateToProps = state => ({
  appoints: state.appoints,
  auth: state.auth,
});

const mapDispatchToProps = dispatch => ({
  storeAppoints: apps => dispatch(SETAPPOINT(apps)),
  delAppoints: app => dispatch(DELAPPOINT(app)),
  setLogout: act => dispatch(SETLOGOUT(act)),
});


const Appointments = ({
  auth, setLogout, appoints, storeAppoints, delAppoints,
}) => {
  const [err, setErr] = useState(null);
  const [isLoaded, setIsLoaded] = useState(false);
  const [alert, setAlert] = useState({ load: false, message: '', type: '' });

  const history = useHistory()
  
  const handleLogout = () => {
    setLogout();
    localStorage.removeItem('tok');
    history.push('/')
  };

  const handleCancel = () => {
    setAlert({ load: false, message: '' });
  };

  const handleConfirmed = () => {
    setAlert({ load: false, message: '' });
  };

  const handleRemoveApp = e => {
    const appId = e.target.dataset.id;
    deleteAppoints(appId, setAlert, delAppoints);
  };

  useEffect(() => {
    if (localStorage.tok) {
      getAppointments(storeAppoints, setIsLoaded, setErr);
      resizer();
    }

    return setIsLoaded(false);
  }, [auth.user, storeAppoints]);


  if (err) {
    return (
      <Error />
    );
  }
  if (!isLoaded) {
    return (
      <div className={`${styles.containerLoad} ${styles.container}`}>
        <img src="https://www.ecoloxtech.com/images/giphycolor.gif" alt="" />
        <h3 data-testid="check-appointment-route">Appointments</h3>
      </div>
    );
  }

  return (
    <div className={`${style.container} ${styles.container}`}>
      <div className={`${style.sideBar} ${styles.sideBar} sideBar`}>
        <h3 data-testid="check-appointment-route">Appointments</h3>
        <div tabIndex="0" role="button" onKeyDown={() => { }} onClick={handleCloseMenu} className={`${styles.closeWrap} ${styles.menuWrap}`}>
          <img src="https://img.icons8.com/color/30/000000/close-window.png" className={styles.closeMenu} alt="" />
        </div>
        <p className={styles.user}>
          <img src="https://img.icons8.com/windows/35/000000/user-male-circle.png" alt="" />
          {' '}
          {auth.user && auth.user.substring(0, 7)}
        </p>
        <Link to="/">
          <img className={style.logo} src={logo} alt="" />
        </Link>
        <div className={`${style.midBar} ${styles.midBar}`}>
          <p>
            {' '}
            <Link to="/doctors">CAREGIVERS</Link>
          </p>
          <p>
            {' '}
            <Link to="/appointments">APPOINTMENTS</Link>
          </p>
        </div>

        <footer className={style.sideFoot}>
          <button type="button" onClick={handleLogout}>Logout</button>
          <p>
            {' '}
            <a href="https://github.com/mikenath223/trixxcare/issues" target="_blank" rel="noopener noreferrer">Help</a>
          </p>
          <p>
            <img src="https://img.icons8.com/android/24/000000/twitter.png" alt="" />
            <img src="https://img.icons8.com/android/24/000000/facebook-new.png" alt="" />
            <img src="https://img.icons8.com/android/24/000000/google-plus.png" alt="" />
            <img src="https://img.icons8.com/material/24/000000/vimeo.png" alt="" />
            <img src="https://img.icons8.com/metro/24/000000/pinterest.png" alt="" />
          </p>
        </footer>
      </div>

      <div className={styles.mainBar}>
        <div role="button" tabIndex="0" onKeyDown={() => { }} className={styles.menuWrap} onClick={handleOpenMenu}>
          <img src="https://img.icons8.com/android/24/000000/hamburger.png" alt="" className={styles.menuIcon} />
        </div>
        <h3>Appointments</h3>
        {appoints.length <= 0 ? (
          <p className={styles.noApp}>
            {' '}
            No appointments yet. Not to worry
            <Link to="/doctors">Browse </Link>
            {' '}
            through our list of caregivers and make an appointment.
          </p>
        ) : null}

        <div className={styles.appBar}>
          <table className={styles.table}>
            <tbody className={styles.tableBody}>
              {appoints.map(app => (
                <tr className={styles.tableRow} key={app.id}>
                  <td className={styles.tableData}>
                    {' '}
                    <img src="https://img.icons8.com/doodle/30/000000/doctor-male.png" alt="" />
                    {app.doctor_name || ''}
                    {' '}
                  </td>
                  <td className={styles.tableData}>
                    <img src="https://img.icons8.com/plasticine/30/000000/time.png" alt="" />
                    {app.date || ''}
                  </td>
                  <td className={styles.tableData}>
                    {app.location}
                  </td>
                  <td className={styles.tableData}>
                    <button type="button" data-id={app.id} className={styles.delButton} onClick={handleRemoveApp}>Cancel Appointment</button>
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
          {alert.type === 'success'
            ? (
              <SweetAlert
                success
                title={alert.message}
                onConfirm={handleConfirmed}
                onCancel={handleCancel}
                timeout={2000}
              />
            )
            : null}
          {alert.type === 'error' ? (
            <SweetAlert
              error
              title={alert.message}
              onConfirm={handleConfirmed}
              onCancel={handleCancel}
              timeout={3500}
            />
          ) : null}
        </div>
      </div>
    </div>
  );
};

Appointments.defaultProps = {
  auth: PropTypes.shape({
    isLogged: false,
    user: '',
  }),
};

Appointments.propTypes = {
  auth: PropTypes.shape({
    isLogged: PropTypes.bool,
    user: PropTypes.string,
  }),
  setLogout: PropTypes.func.isRequired,
  appoints: PropTypes.arrayOf(PropTypes.object).isRequired,
  storeAppoints: PropTypes.func.isRequired,
  delAppoints: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(Appointments);