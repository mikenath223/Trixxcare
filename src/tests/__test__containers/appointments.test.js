import React from 'react';
import withReduxRouter from '../../utils/testUtils';
import Appointments from '../../containers/Appointment/Appointment';

describe('Appointments', function () {
  beforeEach(() => {
    this.props = {
      match: {
        params: {
          name: 'appointments',
        },
      },
    };
  });

  it('should match snapshot', () => {
    const { container } = withReduxRouter(
      <Appointments {...this.props} />,
    );

    expect(container).toMatchSnapshot();
  });
});
