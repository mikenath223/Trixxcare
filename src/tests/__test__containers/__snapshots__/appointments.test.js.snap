// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments should match snapshot 1`] = `
<div>
  <div
    class="containerLoad container"
  >
    <img
      alt=""
      src="https://www.ecoloxtech.com/images/giphycolor.gif"
    />
    <h3
      data-testid="check-appointment-route"
    >
      Appointments
    </h3>
  </div>
</div>
`;
